src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerViewController.swift

```diff
@@ -303,12 +310,18 @@ class QRScannerViewController: UIViewController {
 			previewLayer = AVCaptureVideoPreviewLayer()
 			return
 		}
-		viewModel?.startCaptureSession()
-		
-		previewLayer = AVCaptureVideoPreviewLayer(session: captureSession)
-		previewLayer.frame = view.layer.bounds
-		previewLayer.videoGravity = .resizeAspectFill
-		view.layer.insertSublayer(previewLayer, at: 0)
+		// This extra check of the visible view controller is for Testing purposes only
+		if let viewControllers = navigationController?.viewControllers {
```

Comment:
Perhaps wrap this code in a new function -> startCaptureSession and just call here

Reply:
I was thinking if we could put this loop logic in a function called startCaptureSession and just call that in setupViewModel. I am also fine if you want to stay like this.